diff -u4 -r --new-file include/hip/hip_ext.h modified_include/hip/hip_ext.h
--- include/hip/hip_ext.h	2026-03-27 20:57:07.000000000 +0000
+++ modified_include/hip/hip_ext.h	2026-04-01 22:11:28.000000000 +0000
@@ -69,11 +69,11 @@
                                     uint32_t globalWorkSizeY, uint32_t globalWorkSizeZ,
                                     uint32_t localWorkSizeX, uint32_t localWorkSizeY,
                                     uint32_t localWorkSizeZ, size_t sharedMemBytes,
                                     hipStream_t hStream, void** kernelParams, void** extra,
-                                    hipEvent_t startEvent = nullptr,
-                                    hipEvent_t stopEvent = nullptr,
-                                    uint32_t flags = 0);
+                                    hipEvent_t startEvent,
+                                    hipEvent_t stopEvent,
+                                    uint32_t flags);
 /**
  * @brief This HIP API is deprecated, please use hipExtModuleLaunchKernel() instead.
  *
  */
@@ -82,10 +82,10 @@
                                     uint32_t globalWorkSizeY, uint32_t globalWorkSizeZ,
                                     uint32_t localWorkSizeX, uint32_t localWorkSizeY,
                                     uint32_t localWorkSizeZ, size_t sharedMemBytes,
                                     hipStream_t hStream, void** kernelParams, void** extra,
-                                    hipEvent_t startEvent = nullptr,
-                                    hipEvent_t stopEvent = nullptr)
+                                    hipEvent_t startEvent,
+                                    hipEvent_t stopEvent)
                                     __attribute__((deprecated("use hipExtModuleLaunchKernel instead")));
 
 #if defined(__cplusplus)
 
