diff -u4 -r --new-file include/cuda.h modified_include/cuda.h
--- include/layers/zel_tracing_register_cb.h	2026-02-27 16:26:19.000000000 +0000
+++ modified_include/layers/zel_tracing_register_cb.h	2026-02-27 16:30:52.000000000 +0000
@@ -2727,8 +2727,9 @@
 ///     allowing the callback the ability to modify the parameter's value
 
 typedef struct _zer_get_default_context_params_t
 {
+    uint8_t _dummy; // padding to ensure ABI compatibility between C and C++
 } zer_get_default_context_params_t;
 
 
 ///////////////////////////////////////////////////////////////////////////////
