diff -u4 -r --new-file include/cuda.h modified_include/cuda.h
--- include/cuda.h	2026-03-27 20:57:07.000000000 +0000
+++ modified_include/cuda.h	2026-03-30 22:10:41.000000000 +0000
@@ -290,9 +290,11 @@
 
 /**
  * CUDA IPC handle size
  */
+#ifndef CU_IPC_HANDLE_SIZE
 #define CU_IPC_HANDLE_SIZE 64
+#endif
 
 /**
  * CUDA IPC event handle
  */
@@ -499,9 +501,9 @@
  * Per-operation parameters for ::cuStreamBatchMemOp
  */
 typedef union CUstreamBatchMemOpParams_union {
     CUstreamBatchMemOpType operation;
-    struct CUstreamMemOpWaitValueParams_st {
+    struct {
         CUstreamBatchMemOpType operation;
         CUdeviceptr address;
         union {
             cuuint32_t value;
@@ -509,9 +511,9 @@
         };
         unsigned int flags;
         CUdeviceptr alias; /**< For driver internal use. Initial value is unimportant. */
     } waitValue;
-    struct CUstreamMemOpWriteValueParams_st {
+    struct {
         CUstreamBatchMemOpType operation;
         CUdeviceptr address;
         union {
             cuuint32_t value;
@@ -519,13 +521,13 @@
         };
         unsigned int flags;
         CUdeviceptr alias; /**< For driver internal use. Initial value is unimportant. */
     } writeValue;
-    struct CUstreamMemOpFlushRemoteWritesParams_st {
+    struct {
         CUstreamBatchMemOpType operation;
         unsigned int flags;
     } flushRemoteWrites;
-    struct CUstreamMemOpMemoryBarrierParams_st { /**< Only supported in the _v2 API */
+    struct { /**< Only supported in the _v2 API */
         CUstreamBatchMemOpType operation;
         unsigned int flags;
     } memoryBarrier;
     cuuint64_t pad[6];
@@ -1338,9 +1340,11 @@
 
 /*
  * Indicates that compute device class supports accelerated features.
  */
+#ifndef CU_COMPUTE_ACCELERATED_TARGET_BASE
 #define CU_COMPUTE_ACCELERATED_TARGET_BASE   0x10000
+#endif
 
 /**
  * Online compilation targets
  */
@@ -2851,9 +2855,11 @@
 
 /**
  * Size of tensor map descriptor
  */
+#ifndef CU_TENSOR_MAP_NUM_QWORDS
 #define CU_TENSOR_MAP_NUM_QWORDS 16
+#endif
 
 /**
  * Tensor map descriptor. Requires compiler support for aligning to 64 bytes.
  */
diff -u4 -r --new-file include/cuda_vdpau_interop.h modified_include/cuda_vdpau_interop.h
--- include/cuda_vdpau_interop.h	2026-03-27 20:57:07.000000000 +0000
+++ modified_include/cuda_vdpau_interop.h	2026-03-30 22:10:41.000000000 +0000
@@ -49,11 +49,13 @@
 
 #if !defined(__CUDA_VDPAU_INTEROP_H__)
 #define __CUDA_VDPAU_INTEROP_H__
 
+#ifndef THAPI_NO_INCLUDE
 #include "cuda_runtime_api.h"
 
 #include <vdpau/vdpau.h>
+#endif
 
 #if defined(__cplusplus)
 extern "C" {
 #endif /* __cplusplus */
diff -u4 -r --new-file include/driver_types.h modified_include/driver_types.h
--- include/driver_types.h	2026-03-27 20:57:07.000000000 +0000
+++ modified_include/driver_types.h	2026-03-30 22:10:41.000000000 +0000
@@ -2292,9 +2292,11 @@
 
 /**
  * CUDA IPC Handle Size
  */
+#ifndef CUDA_IPC_HANDLE_SIZE
 #define CUDA_IPC_HANDLE_SIZE 64
+#endif
 
 /**
  * CUDA IPC event handle
  */
