diff -u4 include/cuda.h modified_include/cuda.h
--- include/cuda.h	2020-07-14 13:39:54.180519602 -0500
+++ modified_include/cuda.h	2020-07-14 13:42:16.728001186 -0500
@@ -49,14 +49,12 @@
 
 #ifndef __cuda_cuda_h__
 #define __cuda_cuda_h__
 
-#include <stdlib.h>
 #ifdef _MSC_VER
 typedef unsigned __int32 cuuint32_t;
 typedef unsigned __int64 cuuint64_t;
 #else
-#include <stdint.h>
 typedef uint32_t cuuint32_t;
 typedef uint64_t cuuint64_t;
 #endif
 
