diff -u4 -r --new-file include/cuda.h modified_include/cuda.h
--- include/cuda.h	2023-06-06 10:07:58.570419730 -0500
+++ modified_include/cuda.h	2023-08-02 21:57:02.428626420 -0500
@@ -52,14 +52,12 @@
 
 
 
 
-#include <stdlib.h>
 #ifdef _MSC_VER
 typedef unsigned __int32 cuuint32_t;
 typedef unsigned __int64 cuuint64_t;
 #else
-#include <stdint.h>
 typedef uint32_t cuuint32_t;
 typedef uint64_t cuuint64_t;
 #endif
 
@@ -499,9 +497,9 @@
  * Per-operation parameters for ::cuStreamBatchMemOp
  */
 typedef union CUstreamBatchMemOpParams_union {
     CUstreamBatchMemOpType operation;
-    struct CUstreamMemOpWaitValueParams_st {
+    struct {
         CUstreamBatchMemOpType operation;
         CUdeviceptr address;
         union {
             cuuint32_t value;
@@ -509,9 +507,9 @@
         };
         unsigned int flags;
         CUdeviceptr alias; /**< For driver internal use. Initial value is unimportant. */
     } waitValue;
-    struct CUstreamMemOpWriteValueParams_st {
+    struct {
         CUstreamBatchMemOpType operation;
         CUdeviceptr address;
         union {
             cuuint32_t value;
@@ -519,13 +517,13 @@
         };
         unsigned int flags;
         CUdeviceptr alias; /**< For driver internal use. Initial value is unimportant. */
     } writeValue;
-    struct CUstreamMemOpFlushRemoteWritesParams_st {
+    struct {
         CUstreamBatchMemOpType operation;
         unsigned int flags;
     } flushRemoteWrites;
-    struct CUstreamMemOpMemoryBarrierParams_st { /**< Only supported in the _v2 API */
+    struct { /**< Only supported in the _v2 API */
         CUstreamBatchMemOpType operation;
         unsigned int flags;
     } memoryBarrier;
     cuuint64_t pad[6];
diff -u4 -r --new-file include/cuda_vdpau_interop.h modified_include/cuda_vdpau_interop.h
--- include/cuda_vdpau_interop.h	2023-08-02 21:49:17.225507196 -0500
+++ modified_include/cuda_vdpau_interop.h	2023-08-02 21:57:02.428626420 -0500
@@ -49,11 +49,11 @@
 
 #if !defined(__CUDA_VDPAU_INTEROP_H__)
 #define __CUDA_VDPAU_INTEROP_H__
 
-#include "cuda_runtime_api.h"
+//#include "cuda_runtime_api.h"
 
-#include <vdpau/vdpau.h>
+//#include <vdpau/vdpau.h>
 
 #if defined(__cplusplus)
 extern "C" {
 #endif /* __cplusplus */
