diff -u4 include/cuda.h modified_include/cuda.h
--- include/cuda.h	2020-07-14 13:39:54.180519602 -0500
+++ modified_include/cuda.h	2020-07-15 17:02:39.193569005 -0500
@@ -52,14 +52,12 @@
 
 
 
 
-#include <stdlib.h>
 #ifdef _MSC_VER
 typedef unsigned __int32 cuuint32_t;
 typedef unsigned __int64 cuuint64_t;
 #else
-#include <stdint.h>
 typedef uint32_t cuuint32_t;
 typedef uint64_t cuuint64_t;
 #endif
 
@@ -499,9 +497,9 @@
  * Per-operation parameters for ::cuStreamBatchMemOp
  */
 typedef union CUstreamBatchMemOpParams_union {
     CUstreamBatchMemOpType operation;
-    struct CUstreamMemOpWaitValueParams_st {
+    struct {
         CUstreamBatchMemOpType operation;
         CUdeviceptr address;
         union {
             cuuint32_t value;
@@ -509,9 +507,9 @@
         };
         unsigned int flags;
         CUdeviceptr alias; /**< For driver internal use. Initial value is unimportant. */
     } waitValue;
-    struct CUstreamMemOpWriteValueParams_st {
+    struct {
         CUstreamBatchMemOpType operation;
         CUdeviceptr address;
         union {
             cuuint32_t value;
@@ -523,13 +521,13 @@
         };
         unsigned int flags;
         CUdeviceptr alias; /**< For driver internal use. Initial value is unimportant. */
     } writeValue;
-    struct CUstreamMemOpFlushRemoteWritesParams_st {
+    struct {
         CUstreamBatchMemOpType operation;
         unsigned int flags;
     } flushRemoteWrites;
-    struct CUstreamMemOpMemoryBarrierParams_st { /**< Only supported in the _v2 API */
+    struct { /**< Only supported in the _v2 API */
         CUstreamBatchMemOpType operation;
         unsigned int flags;
     } memoryBarrier;
     cuuint64_t pad[6];
