diff -u4 -r include/layers/zel_tracing_api.h modified_include/layers/zel_tracing_api.h
--- include/layers/zel_tracing_api.h	2021-06-09 20:31:07.139669164 -0500
+++ modified_include/layers/zel_tracing_api.h	2021-06-17 17:12:58.581078073 -0500
@@ -54,9 +54,9 @@
 {
     ZEL_STRUCTURE_TYPE_TRACER_DESC = 0x1  ,///< ::zel_tracer_desc_t
     // This enumeration value is deprecated.
     // Pluse use ZEL_STRUCTURE_TYPE_TRACER_DESC.
-    ZEL_STRUCTURE_TYPE_TRACER_EXP_DESC = 0x1  ,///< ::zel_tracer_desc_t
+    //ZEL_STRUCTURE_TYPE_TRACER_EXP_DESC = 0x1  ,///< ::zel_tracer_desc_t
     ZEL_STRUCTURE_TYPE_FORCE_UINT32 = 0x7fffffff
 
 } zel_structure_type_t;
 
diff -u4 -r include/layers/zel_tracing_register_cb.h modified_include/layers/zel_tracing_register_cb.h
--- include/layers/zel_tracing_register_cb.h	2021-06-17 16:28:11.692143702 -0500
+++ modified_include/layers/zel_tracing_register_cb.h	2021-06-17 17:12:58.581078073 -0500
@@ -21,9 +21,9 @@
 #endif
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Handle of tracer object
-typedef struct _zel_tracer_handle_t *zel_tracer_handle_t;
+//typedef struct _zel_tracer_handle_t *zel_tracer_handle_t;
 
 /// Callback definitions for all API released in LevelZero spec 1.1 or newer
 /// Callbacks for APIs included in spec 1.0 are contained in ze_api.helper
 
diff -u4 -r include/loader/ze_loader_api.h modified_include/loader/ze_loader_api.h
--- include/loader/ze_loader_api.h	2021-04-14 14:00:18.987910312 -0500
+++ modified_include/loader/ze_loader_api.h	2021-06-17 17:12:58.581078073 -0500
@@ -7,9 +7,9 @@
  * @file ze_loader_api.cpp
  *
  */
 
-#include "ze_loader.h"
+#include "../ze_api.h"
 
 #if defined(__cplusplus)
 extern "C" {
 #endif
diff -u4 -r include/ze_api.h modified_include/ze_api.h
--- include/ze_api.h	2021-06-09 20:31:07.139669164 -0500
+++ modified_include/ze_api.h	2021-06-17 17:12:58.581078073 -0500
@@ -14,10 +14,10 @@
 #pragma once
 #endif
 
 // standard headers
-#include <stdint.h>
-#include <stddef.h>
+//#include <stdint.h>
+//#include <stddef.h>
 
 #if defined(__cplusplus)
 extern "C" {
 #endif
@@ -26,24 +26,24 @@
 #if !defined(__GNUC__)
 #pragma region common
 #endif
 ///////////////////////////////////////////////////////////////////////////////
-#ifndef ZE_MAKE_VERSION
+//#ifndef ZE_MAKE_VERSION
 /// @brief Generates generic 'oneAPI' API versions
-#define ZE_MAKE_VERSION( _major, _minor )  (( _major << 16 )|( _minor & 0x0000ffff))
-#endif // ZE_MAKE_VERSION
+//#define ZE_MAKE_VERSION( _major, _minor )  (( _major << 16 )|( _minor & 0x0000ffff))
+//#endif // ZE_MAKE_VERSION
 
 ///////////////////////////////////////////////////////////////////////////////
-#ifndef ZE_MAJOR_VERSION
+//#ifndef ZE_MAJOR_VERSION
 /// @brief Extracts 'oneAPI' API major version
-#define ZE_MAJOR_VERSION( _ver )  ( _ver >> 16 )
-#endif // ZE_MAJOR_VERSION
+//#define ZE_MAJOR_VERSION( _ver )  ( _ver >> 16 )
+//#endif // ZE_MAJOR_VERSION
 
 ///////////////////////////////////////////////////////////////////////////////
-#ifndef ZE_MINOR_VERSION
+//#ifndef ZE_MINOR_VERSION
 /// @brief Extracts 'oneAPI' API minor version
-#define ZE_MINOR_VERSION( _ver )  ( _ver & 0x0000ffff )
-#endif // ZE_MINOR_VERSION
+//#define ZE_MINOR_VERSION( _ver )  ( _ver & 0x0000ffff )
+//#endif // ZE_MINOR_VERSION
 
 ///////////////////////////////////////////////////////////////////////////////
 #ifndef ZE_APICALL
 #if defined(_WIN32)
@@ -164,12 +164,12 @@
 
 } ze_ipc_event_pool_handle_t;
 
 ///////////////////////////////////////////////////////////////////////////////
-#ifndef ZE_BIT
+//#ifndef ZE_BIT
 /// @brief Generic macro for enumerator bit masks
-#define ZE_BIT( _i )  ( 1 << _i )
-#endif // ZE_BIT
+//#define ZE_BIT( _i )  ( 1 << _i )
+//#endif // ZE_BIT
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Defines Return/Error codes
 typedef enum _ze_result_t
@@ -308,221 +308,221 @@
 /// @brief Forces all shared allocations into device memory
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_ipc_mem_handle_t
-typedef struct _ze_ipc_mem_handle_t ze_ipc_mem_handle_t;
+//typedef struct _ze_ipc_mem_handle_t ze_ipc_mem_handle_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_ipc_event_pool_handle_t
-typedef struct _ze_ipc_event_pool_handle_t ze_ipc_event_pool_handle_t;
+//typedef struct _ze_ipc_event_pool_handle_t ze_ipc_event_pool_handle_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_base_properties_t
-typedef struct _ze_base_properties_t ze_base_properties_t;
+//typedef struct _ze_base_properties_t ze_base_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_base_desc_t
-typedef struct _ze_base_desc_t ze_base_desc_t;
+//typedef struct _ze_base_desc_t ze_base_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_driver_uuid_t
-typedef struct _ze_driver_uuid_t ze_driver_uuid_t;
+//typedef struct _ze_driver_uuid_t ze_driver_uuid_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_driver_properties_t
-typedef struct _ze_driver_properties_t ze_driver_properties_t;
+//typedef struct _ze_driver_properties_t ze_driver_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_driver_ipc_properties_t
-typedef struct _ze_driver_ipc_properties_t ze_driver_ipc_properties_t;
+//typedef struct _ze_driver_ipc_properties_t ze_driver_ipc_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_driver_extension_properties_t
-typedef struct _ze_driver_extension_properties_t ze_driver_extension_properties_t;
+//typedef struct _ze_driver_extension_properties_t ze_driver_extension_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_uuid_t
-typedef struct _ze_device_uuid_t ze_device_uuid_t;
+//typedef struct _ze_device_uuid_t ze_device_uuid_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_properties_t
-typedef struct _ze_device_properties_t ze_device_properties_t;
+//typedef struct _ze_device_properties_t ze_device_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_thread_t
-typedef struct _ze_device_thread_t ze_device_thread_t;
+//typedef struct _ze_device_thread_t ze_device_thread_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_compute_properties_t
-typedef struct _ze_device_compute_properties_t ze_device_compute_properties_t;
+//typedef struct _ze_device_compute_properties_t ze_device_compute_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_native_kernel_uuid_t
-typedef struct _ze_native_kernel_uuid_t ze_native_kernel_uuid_t;
+//typedef struct _ze_native_kernel_uuid_t ze_native_kernel_uuid_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_module_properties_t
-typedef struct _ze_device_module_properties_t ze_device_module_properties_t;
+//typedef struct _ze_device_module_properties_t ze_device_module_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_command_queue_group_properties_t
-typedef struct _ze_command_queue_group_properties_t ze_command_queue_group_properties_t;
+//typedef struct _ze_command_queue_group_properties_t ze_command_queue_group_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_memory_properties_t
-typedef struct _ze_device_memory_properties_t ze_device_memory_properties_t;
+//typedef struct _ze_device_memory_properties_t ze_device_memory_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_memory_access_properties_t
-typedef struct _ze_device_memory_access_properties_t ze_device_memory_access_properties_t;
+//typedef struct _ze_device_memory_access_properties_t ze_device_memory_access_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_cache_properties_t
-typedef struct _ze_device_cache_properties_t ze_device_cache_properties_t;
+//typedef struct _ze_device_cache_properties_t ze_device_cache_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_image_properties_t
-typedef struct _ze_device_image_properties_t ze_device_image_properties_t;
+//typedef struct _ze_device_image_properties_t ze_device_image_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_external_memory_properties_t
-typedef struct _ze_device_external_memory_properties_t ze_device_external_memory_properties_t;
+//typedef struct _ze_device_external_memory_properties_t ze_device_external_memory_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_p2p_properties_t
-typedef struct _ze_device_p2p_properties_t ze_device_p2p_properties_t;
+//typedef struct _ze_device_p2p_properties_t ze_device_p2p_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_context_desc_t
-typedef struct _ze_context_desc_t ze_context_desc_t;
+//typedef struct _ze_context_desc_t ze_context_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_command_queue_desc_t
-typedef struct _ze_command_queue_desc_t ze_command_queue_desc_t;
+//typedef struct _ze_command_queue_desc_t ze_command_queue_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_command_list_desc_t
-typedef struct _ze_command_list_desc_t ze_command_list_desc_t;
+//typedef struct _ze_command_list_desc_t ze_command_list_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_copy_region_t
-typedef struct _ze_copy_region_t ze_copy_region_t;
+//typedef struct _ze_copy_region_t ze_copy_region_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_image_region_t
-typedef struct _ze_image_region_t ze_image_region_t;
+//typedef struct _ze_image_region_t ze_image_region_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_event_pool_desc_t
-typedef struct _ze_event_pool_desc_t ze_event_pool_desc_t;
+//typedef struct _ze_event_pool_desc_t ze_event_pool_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_event_desc_t
-typedef struct _ze_event_desc_t ze_event_desc_t;
+//typedef struct _ze_event_desc_t ze_event_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_kernel_timestamp_data_t
-typedef struct _ze_kernel_timestamp_data_t ze_kernel_timestamp_data_t;
+//typedef struct _ze_kernel_timestamp_data_t ze_kernel_timestamp_data_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_kernel_timestamp_result_t
-typedef struct _ze_kernel_timestamp_result_t ze_kernel_timestamp_result_t;
+//typedef struct _ze_kernel_timestamp_result_t ze_kernel_timestamp_result_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_fence_desc_t
-typedef struct _ze_fence_desc_t ze_fence_desc_t;
+//typedef struct _ze_fence_desc_t ze_fence_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_image_format_t
-typedef struct _ze_image_format_t ze_image_format_t;
+//typedef struct _ze_image_format_t ze_image_format_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_image_desc_t
-typedef struct _ze_image_desc_t ze_image_desc_t;
+//typedef struct _ze_image_desc_t ze_image_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_image_properties_t
-typedef struct _ze_image_properties_t ze_image_properties_t;
+//typedef struct _ze_image_properties_t ze_image_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_mem_alloc_desc_t
-typedef struct _ze_device_mem_alloc_desc_t ze_device_mem_alloc_desc_t;
+//typedef struct _ze_device_mem_alloc_desc_t ze_device_mem_alloc_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_host_mem_alloc_desc_t
-typedef struct _ze_host_mem_alloc_desc_t ze_host_mem_alloc_desc_t;
+//typedef struct _ze_host_mem_alloc_desc_t ze_host_mem_alloc_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_memory_allocation_properties_t
-typedef struct _ze_memory_allocation_properties_t ze_memory_allocation_properties_t;
+//typedef struct _ze_memory_allocation_properties_t ze_memory_allocation_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_external_memory_export_desc_t
-typedef struct _ze_external_memory_export_desc_t ze_external_memory_export_desc_t;
+//typedef struct _ze_external_memory_export_desc_t ze_external_memory_export_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_external_memory_import_fd_t
-typedef struct _ze_external_memory_import_fd_t ze_external_memory_import_fd_t;
+//typedef struct _ze_external_memory_import_fd_t ze_external_memory_import_fd_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_external_memory_export_fd_t
-typedef struct _ze_external_memory_export_fd_t ze_external_memory_export_fd_t;
+//typedef struct _ze_external_memory_export_fd_t ze_external_memory_export_fd_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_module_constants_t
-typedef struct _ze_module_constants_t ze_module_constants_t;
+//typedef struct _ze_module_constants_t ze_module_constants_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_module_desc_t
-typedef struct _ze_module_desc_t ze_module_desc_t;
+//typedef struct _ze_module_desc_t ze_module_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_module_properties_t
-typedef struct _ze_module_properties_t ze_module_properties_t;
+//typedef struct _ze_module_properties_t ze_module_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_kernel_desc_t
-typedef struct _ze_kernel_desc_t ze_kernel_desc_t;
+//typedef struct _ze_kernel_desc_t ze_kernel_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_kernel_uuid_t
-typedef struct _ze_kernel_uuid_t ze_kernel_uuid_t;
+//typedef struct _ze_kernel_uuid_t ze_kernel_uuid_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_kernel_properties_t
-typedef struct _ze_kernel_properties_t ze_kernel_properties_t;
+//typedef struct _ze_kernel_properties_t ze_kernel_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_group_count_t
-typedef struct _ze_group_count_t ze_group_count_t;
+//typedef struct _ze_group_count_t ze_group_count_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_module_program_exp_desc_t
-typedef struct _ze_module_program_exp_desc_t ze_module_program_exp_desc_t;
+//typedef struct _ze_module_program_exp_desc_t ze_module_program_exp_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_raytracing_ext_properties_t
-typedef struct _ze_device_raytracing_ext_properties_t ze_device_raytracing_ext_properties_t;
+//typedef struct _ze_device_raytracing_ext_properties_t ze_device_raytracing_ext_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_raytracing_mem_alloc_ext_desc_t
-typedef struct _ze_raytracing_mem_alloc_ext_desc_t ze_raytracing_mem_alloc_ext_desc_t;
+//typedef struct _ze_raytracing_mem_alloc_ext_desc_t ze_raytracing_mem_alloc_ext_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_sampler_desc_t
-typedef struct _ze_sampler_desc_t ze_sampler_desc_t;
+//typedef struct _ze_sampler_desc_t ze_sampler_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_physical_mem_desc_t
-typedef struct _ze_physical_mem_desc_t ze_physical_mem_desc_t;
+//typedef struct _ze_physical_mem_desc_t ze_physical_mem_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_float_atomic_ext_properties_t
-typedef struct _ze_float_atomic_ext_properties_t ze_float_atomic_ext_properties_t;
+//typedef struct _ze_float_atomic_ext_properties_t ze_float_atomic_ext_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_relaxed_allocation_limits_exp_desc_t
-typedef struct _ze_relaxed_allocation_limits_exp_desc_t ze_relaxed_allocation_limits_exp_desc_t;
+//typedef struct _ze_relaxed_allocation_limits_exp_desc_t ze_relaxed_allocation_limits_exp_desc_t;
 
 
 #if !defined(__GNUC__)
 #pragma endregion
diff -u4 -r include/zes_api.h modified_include/zes_api.h
--- include/zes_api.h	2021-06-09 20:31:07.139669164 -0500
+++ modified_include/zes_api.h	2021-06-17 17:12:58.581078073 -0500
@@ -182,249 +182,249 @@
 } zes_base_capability_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_base_properties_t
-typedef struct _zes_base_properties_t zes_base_properties_t;
+//typedef struct _zes_base_properties_t zes_base_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_base_desc_t
-typedef struct _zes_base_desc_t zes_base_desc_t;
+//typedef struct _zes_base_desc_t zes_base_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_base_state_t
-typedef struct _zes_base_state_t zes_base_state_t;
+//typedef struct _zes_base_state_t zes_base_state_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_base_config_t
-typedef struct _zes_base_config_t zes_base_config_t;
+//typedef struct _zes_base_config_t zes_base_config_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_base_capability_t
-typedef struct _zes_base_capability_t zes_base_capability_t;
+//typedef struct _zes_base_capability_t zes_base_capability_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_device_state_t
-typedef struct _zes_device_state_t zes_device_state_t;
+//typedef struct _zes_device_state_t zes_device_state_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_device_properties_t
-typedef struct _zes_device_properties_t zes_device_properties_t;
+//typedef struct _zes_device_properties_t zes_device_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_process_state_t
-typedef struct _zes_process_state_t zes_process_state_t;
+//typedef struct _zes_process_state_t zes_process_state_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_pci_address_t
-typedef struct _zes_pci_address_t zes_pci_address_t;
+//typedef struct _zes_pci_address_t zes_pci_address_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_pci_speed_t
-typedef struct _zes_pci_speed_t zes_pci_speed_t;
+//typedef struct _zes_pci_speed_t zes_pci_speed_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_pci_properties_t
-typedef struct _zes_pci_properties_t zes_pci_properties_t;
+//typedef struct _zes_pci_properties_t zes_pci_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_pci_state_t
-typedef struct _zes_pci_state_t zes_pci_state_t;
+//typedef struct _zes_pci_state_t zes_pci_state_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_pci_bar_properties_t
-typedef struct _zes_pci_bar_properties_t zes_pci_bar_properties_t;
+//typedef struct _zes_pci_bar_properties_t zes_pci_bar_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_pci_stats_t
-typedef struct _zes_pci_stats_t zes_pci_stats_t;
+//typedef struct _zes_pci_stats_t zes_pci_stats_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_diag_test_t
-typedef struct _zes_diag_test_t zes_diag_test_t;
+//typedef struct _zes_diag_test_t zes_diag_test_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_diag_properties_t
-typedef struct _zes_diag_properties_t zes_diag_properties_t;
+//typedef struct _zes_diag_properties_t zes_diag_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_engine_properties_t
-typedef struct _zes_engine_properties_t zes_engine_properties_t;
+//typedef struct _zes_engine_properties_t zes_engine_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_engine_stats_t
-typedef struct _zes_engine_stats_t zes_engine_stats_t;
+//typedef struct _zes_engine_stats_t zes_engine_stats_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_fabric_port_id_t
-typedef struct _zes_fabric_port_id_t zes_fabric_port_id_t;
+//typedef struct _zes_fabric_port_id_t zes_fabric_port_id_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_fabric_port_speed_t
-typedef struct _zes_fabric_port_speed_t zes_fabric_port_speed_t;
+//typedef struct _zes_fabric_port_speed_t zes_fabric_port_speed_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_fabric_port_properties_t
-typedef struct _zes_fabric_port_properties_t zes_fabric_port_properties_t;
+//typedef struct _zes_fabric_port_properties_t zes_fabric_port_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_fabric_link_type_t
-typedef struct _zes_fabric_link_type_t zes_fabric_link_type_t;
+//typedef struct _zes_fabric_link_type_t zes_fabric_link_type_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_fabric_port_config_t
-typedef struct _zes_fabric_port_config_t zes_fabric_port_config_t;
+//typedef struct _zes_fabric_port_config_t zes_fabric_port_config_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_fabric_port_state_t
-typedef struct _zes_fabric_port_state_t zes_fabric_port_state_t;
+//typedef struct _zes_fabric_port_state_t zes_fabric_port_state_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_fabric_port_throughput_t
-typedef struct _zes_fabric_port_throughput_t zes_fabric_port_throughput_t;
+//typedef struct _zes_fabric_port_throughput_t zes_fabric_port_throughput_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_fan_speed_t
-typedef struct _zes_fan_speed_t zes_fan_speed_t;
+//typedef struct _zes_fan_speed_t zes_fan_speed_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_fan_temp_speed_t
-typedef struct _zes_fan_temp_speed_t zes_fan_temp_speed_t;
+//typedef struct _zes_fan_temp_speed_t zes_fan_temp_speed_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_fan_speed_table_t
-typedef struct _zes_fan_speed_table_t zes_fan_speed_table_t;
+//typedef struct _zes_fan_speed_table_t zes_fan_speed_table_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_fan_properties_t
-typedef struct _zes_fan_properties_t zes_fan_properties_t;
+//typedef struct _zes_fan_properties_t zes_fan_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_fan_config_t
-typedef struct _zes_fan_config_t zes_fan_config_t;
+//typedef struct _zes_fan_config_t zes_fan_config_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_firmware_properties_t
-typedef struct _zes_firmware_properties_t zes_firmware_properties_t;
+//typedef struct _zes_firmware_properties_t zes_firmware_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_freq_properties_t
-typedef struct _zes_freq_properties_t zes_freq_properties_t;
+//typedef struct _zes_freq_properties_t zes_freq_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_freq_range_t
-typedef struct _zes_freq_range_t zes_freq_range_t;
+//typedef struct _zes_freq_range_t zes_freq_range_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_freq_state_t
-typedef struct _zes_freq_state_t zes_freq_state_t;
+//typedef struct _zes_freq_state_t zes_freq_state_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_freq_throttle_time_t
-typedef struct _zes_freq_throttle_time_t zes_freq_throttle_time_t;
+//typedef struct _zes_freq_throttle_time_t zes_freq_throttle_time_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_oc_capabilities_t
-typedef struct _zes_oc_capabilities_t zes_oc_capabilities_t;
+//typedef struct _zes_oc_capabilities_t zes_oc_capabilities_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_led_properties_t
-typedef struct _zes_led_properties_t zes_led_properties_t;
+//typedef struct _zes_led_properties_t zes_led_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_led_color_t
-typedef struct _zes_led_color_t zes_led_color_t;
+//typedef struct _zes_led_color_t zes_led_color_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_led_state_t
-typedef struct _zes_led_state_t zes_led_state_t;
+//typedef struct _zes_led_state_t zes_led_state_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_mem_properties_t
-typedef struct _zes_mem_properties_t zes_mem_properties_t;
+//typedef struct _zes_mem_properties_t zes_mem_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_mem_state_t
-typedef struct _zes_mem_state_t zes_mem_state_t;
+//typedef struct _zes_mem_state_t zes_mem_state_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_mem_bandwidth_t
-typedef struct _zes_mem_bandwidth_t zes_mem_bandwidth_t;
+//typedef struct _zes_mem_bandwidth_t zes_mem_bandwidth_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_perf_properties_t
-typedef struct _zes_perf_properties_t zes_perf_properties_t;
+//typedef struct _zes_perf_properties_t zes_perf_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_power_properties_t
-typedef struct _zes_power_properties_t zes_power_properties_t;
+//typedef struct _zes_power_properties_t zes_power_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_power_energy_counter_t
-typedef struct _zes_power_energy_counter_t zes_power_energy_counter_t;
+//typedef struct _zes_power_energy_counter_t zes_power_energy_counter_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_power_sustained_limit_t
-typedef struct _zes_power_sustained_limit_t zes_power_sustained_limit_t;
+//typedef struct _zes_power_sustained_limit_t zes_power_sustained_limit_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_power_burst_limit_t
-typedef struct _zes_power_burst_limit_t zes_power_burst_limit_t;
+//typedef struct _zes_power_burst_limit_t zes_power_burst_limit_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_power_peak_limit_t
-typedef struct _zes_power_peak_limit_t zes_power_peak_limit_t;
+//typedef struct _zes_power_peak_limit_t zes_power_peak_limit_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_energy_threshold_t
-typedef struct _zes_energy_threshold_t zes_energy_threshold_t;
+//typedef struct _zes_energy_threshold_t zes_energy_threshold_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_psu_properties_t
-typedef struct _zes_psu_properties_t zes_psu_properties_t;
+//typedef struct _zes_psu_properties_t zes_psu_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_psu_state_t
-typedef struct _zes_psu_state_t zes_psu_state_t;
+//typedef struct _zes_psu_state_t zes_psu_state_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_ras_properties_t
-typedef struct _zes_ras_properties_t zes_ras_properties_t;
+//typedef struct _zes_ras_properties_t zes_ras_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_ras_state_t
-typedef struct _zes_ras_state_t zes_ras_state_t;
+//typedef struct _zes_ras_state_t zes_ras_state_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_ras_config_t
-typedef struct _zes_ras_config_t zes_ras_config_t;
+//typedef struct _zes_ras_config_t zes_ras_config_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_sched_properties_t
-typedef struct _zes_sched_properties_t zes_sched_properties_t;
+//typedef struct _zes_sched_properties_t zes_sched_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_sched_timeout_properties_t
-typedef struct _zes_sched_timeout_properties_t zes_sched_timeout_properties_t;
+//typedef struct _zes_sched_timeout_properties_t zes_sched_timeout_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_sched_timeslice_properties_t
-typedef struct _zes_sched_timeslice_properties_t zes_sched_timeslice_properties_t;
+//typedef struct _zes_sched_timeslice_properties_t zes_sched_timeslice_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_standby_properties_t
-typedef struct _zes_standby_properties_t zes_standby_properties_t;
+//typedef struct _zes_standby_properties_t zes_standby_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_temp_properties_t
-typedef struct _zes_temp_properties_t zes_temp_properties_t;
+//typedef struct _zes_temp_properties_t zes_temp_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_temp_threshold_t
-typedef struct _zes_temp_threshold_t zes_temp_threshold_t;
+//typedef struct _zes_temp_threshold_t zes_temp_threshold_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zes_temp_config_t
-typedef struct _zes_temp_config_t zes_temp_config_t;
+//typedef struct _zes_temp_config_t zes_temp_config_t;
 
 
 #if !defined(__GNUC__)
 #pragma endregion
diff -u4 -r include/zet_api.h modified_include/zet_api.h
--- include/zet_api.h	2021-06-09 20:31:07.143669175 -0500
+++ modified_include/zet_api.h	2021-06-17 17:12:58.581078073 -0500
@@ -147,85 +147,85 @@
 } zet_typed_value_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_base_properties_t
-typedef struct _zet_base_properties_t zet_base_properties_t;
+//typedef struct _zet_base_properties_t zet_base_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_base_desc_t
-typedef struct _zet_base_desc_t zet_base_desc_t;
+//typedef struct _zet_base_desc_t zet_base_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_typed_value_t
-typedef struct _zet_typed_value_t zet_typed_value_t;
+//typedef struct _zet_typed_value_t zet_typed_value_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_device_debug_properties_t
-typedef struct _zet_device_debug_properties_t zet_device_debug_properties_t;
+//typedef struct _zet_device_debug_properties_t zet_device_debug_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_debug_config_t
-typedef struct _zet_debug_config_t zet_debug_config_t;
+//typedef struct _zet_debug_config_t zet_debug_config_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_debug_event_info_detached_t
-typedef struct _zet_debug_event_info_detached_t zet_debug_event_info_detached_t;
+//typedef struct _zet_debug_event_info_detached_t zet_debug_event_info_detached_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_debug_event_info_module_t
-typedef struct _zet_debug_event_info_module_t zet_debug_event_info_module_t;
+//typedef struct _zet_debug_event_info_module_t zet_debug_event_info_module_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_debug_event_info_thread_stopped_t
-typedef struct _zet_debug_event_info_thread_stopped_t zet_debug_event_info_thread_stopped_t;
+//typedef struct _zet_debug_event_info_thread_stopped_t zet_debug_event_info_thread_stopped_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_debug_event_info_page_fault_t
-typedef struct _zet_debug_event_info_page_fault_t zet_debug_event_info_page_fault_t;
+//typedef struct _zet_debug_event_info_page_fault_t zet_debug_event_info_page_fault_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_debug_event_t
-typedef struct _zet_debug_event_t zet_debug_event_t;
+//typedef struct _zet_debug_event_t zet_debug_event_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_debug_memory_space_desc_t
-typedef struct _zet_debug_memory_space_desc_t zet_debug_memory_space_desc_t;
+//typedef struct _zet_debug_memory_space_desc_t zet_debug_memory_space_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_debug_regset_properties_t
-typedef struct _zet_debug_regset_properties_t zet_debug_regset_properties_t;
+//typedef struct _zet_debug_regset_properties_t zet_debug_regset_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_metric_group_properties_t
-typedef struct _zet_metric_group_properties_t zet_metric_group_properties_t;
+//typedef struct _zet_metric_group_properties_t zet_metric_group_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_metric_properties_t
-typedef struct _zet_metric_properties_t zet_metric_properties_t;
+//typedef struct _zet_metric_properties_t zet_metric_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_metric_streamer_desc_t
-typedef struct _zet_metric_streamer_desc_t zet_metric_streamer_desc_t;
+//typedef struct _zet_metric_streamer_desc_t zet_metric_streamer_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_metric_query_pool_desc_t
-typedef struct _zet_metric_query_pool_desc_t zet_metric_query_pool_desc_t;
+//typedef struct _zet_metric_query_pool_desc_t zet_metric_query_pool_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_profile_properties_t
-typedef struct _zet_profile_properties_t zet_profile_properties_t;
+//typedef struct _zet_profile_properties_t zet_profile_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_profile_free_register_token_t
-typedef struct _zet_profile_free_register_token_t zet_profile_free_register_token_t;
+//typedef struct _zet_profile_free_register_token_t zet_profile_free_register_token_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_profile_register_sequence_t
-typedef struct _zet_profile_register_sequence_t zet_profile_register_sequence_t;
+//typedef struct _zet_profile_register_sequence_t zet_profile_register_sequence_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_tracer_exp_desc_t
-typedef struct _zet_tracer_exp_desc_t zet_tracer_exp_desc_t;
+//typedef struct _zet_tracer_exp_desc_t zet_tracer_exp_desc_t;
 
 
 #if !defined(__GNUC__)
 #pragma endregion
