diff -u4 -r include/ze_api.h modified_include/ze_api.h
--- include/ze_api.h	2020-08-03 15:26:37.052772417 -0500
+++ modified_include/ze_api.h	2020-08-03 15:52:15.495493149 -0500
@@ -14,10 +14,10 @@
 #pragma once
 #endif
 
 // standard headers
-#include <stdint.h>
-#include <stddef.h>
+//#include <stdint.h>
+//#include <stddef.h>
 
 #if defined(__cplusplus)
 extern "C" {
 #endif
@@ -26,24 +26,24 @@
 #if !defined(__GNUC__)
 #pragma region common
 #endif
 ///////////////////////////////////////////////////////////////////////////////
-#ifndef ZE_MAKE_VERSION
+//#ifndef ZE_MAKE_VERSION
 /// @brief Generates generic 'oneAPI' API versions
-#define ZE_MAKE_VERSION( _major, _minor )  (( _major << 16 )|( _minor & 0x0000ffff))
-#endif // ZE_MAKE_VERSION
+//#define ZE_MAKE_VERSION( _major, _minor )  (( _major << 16 )|( _minor & 0x0000ffff))
+//#endif // ZE_MAKE_VERSION
 
 ///////////////////////////////////////////////////////////////////////////////
-#ifndef ZE_MAJOR_VERSION
+//#ifndef ZE_MAJOR_VERSION
 /// @brief Extracts 'oneAPI' API major version
-#define ZE_MAJOR_VERSION( _ver )  ( _ver >> 16 )
-#endif // ZE_MAJOR_VERSION
+//#define ZE_MAJOR_VERSION( _ver )  ( _ver >> 16 )
+//#endif // ZE_MAJOR_VERSION
 
 ///////////////////////////////////////////////////////////////////////////////
-#ifndef ZE_MINOR_VERSION
+//#ifndef ZE_MINOR_VERSION
 /// @brief Extracts 'oneAPI' API minor version
-#define ZE_MINOR_VERSION( _ver )  ( _ver & 0x0000ffff )
-#endif // ZE_MINOR_VERSION
+//#define ZE_MINOR_VERSION( _ver )  ( _ver & 0x0000ffff )
+//#endif // ZE_MINOR_VERSION
 
 ///////////////////////////////////////////////////////////////////////////////
 #ifndef ZE_APICALL
 #if defined(_WIN32)
@@ -164,12 +164,12 @@
 
 } ze_ipc_event_pool_handle_t;
 
 ///////////////////////////////////////////////////////////////////////////////
-#ifndef ZE_BIT
+//#ifndef ZE_BIT
 /// @brief Generic macro for enumerator bit masks
-#define ZE_BIT( _i )  ( 1 << _i )
-#endif // ZE_BIT
+//#define ZE_BIT( _i )  ( 1 << _i )
+//#endif // ZE_BIT
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Defines Return/Error codes
 typedef enum _ze_result_t
@@ -304,205 +304,205 @@
 /// @brief Forces all shared allocations into device memory
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_ipc_mem_handle_t
-typedef struct _ze_ipc_mem_handle_t ze_ipc_mem_handle_t;
+//typedef struct _ze_ipc_mem_handle_t ze_ipc_mem_handle_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_ipc_event_pool_handle_t
-typedef struct _ze_ipc_event_pool_handle_t ze_ipc_event_pool_handle_t;
+//typedef struct _ze_ipc_event_pool_handle_t ze_ipc_event_pool_handle_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_base_properties_t
-typedef struct _ze_base_properties_t ze_base_properties_t;
+//typedef struct _ze_base_properties_t ze_base_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_base_desc_t
-typedef struct _ze_base_desc_t ze_base_desc_t;
+//typedef struct _ze_base_desc_t ze_base_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_driver_uuid_t
-typedef struct _ze_driver_uuid_t ze_driver_uuid_t;
+//typedef struct _ze_driver_uuid_t ze_driver_uuid_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_driver_properties_t
-typedef struct _ze_driver_properties_t ze_driver_properties_t;
+//typedef struct _ze_driver_properties_t ze_driver_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_driver_ipc_properties_t
-typedef struct _ze_driver_ipc_properties_t ze_driver_ipc_properties_t;
+//typedef struct _ze_driver_ipc_properties_t ze_driver_ipc_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_driver_extension_properties_t
-typedef struct _ze_driver_extension_properties_t ze_driver_extension_properties_t;
+//typedef struct _ze_driver_extension_properties_t ze_driver_extension_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_uuid_t
-typedef struct _ze_device_uuid_t ze_device_uuid_t;
+//typedef struct _ze_device_uuid_t ze_device_uuid_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_properties_t
-typedef struct _ze_device_properties_t ze_device_properties_t;
+//typedef struct _ze_device_properties_t ze_device_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_thread_t
-typedef struct _ze_device_thread_t ze_device_thread_t;
+//typedef struct _ze_device_thread_t ze_device_thread_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_compute_properties_t
-typedef struct _ze_device_compute_properties_t ze_device_compute_properties_t;
+//typedef struct _ze_device_compute_properties_t ze_device_compute_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_native_kernel_uuid_t
-typedef struct _ze_native_kernel_uuid_t ze_native_kernel_uuid_t;
+//typedef struct _ze_native_kernel_uuid_t ze_native_kernel_uuid_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_module_properties_t
-typedef struct _ze_device_module_properties_t ze_device_module_properties_t;
+//typedef struct _ze_device_module_properties_t ze_device_module_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_command_queue_group_properties_t
-typedef struct _ze_command_queue_group_properties_t ze_command_queue_group_properties_t;
+//typedef struct _ze_command_queue_group_properties_t ze_command_queue_group_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_memory_properties_t
-typedef struct _ze_device_memory_properties_t ze_device_memory_properties_t;
+//typedef struct _ze_device_memory_properties_t ze_device_memory_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_memory_access_properties_t
-typedef struct _ze_device_memory_access_properties_t ze_device_memory_access_properties_t;
+//typedef struct _ze_device_memory_access_properties_t ze_device_memory_access_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_cache_properties_t
-typedef struct _ze_device_cache_properties_t ze_device_cache_properties_t;
+//typedef struct _ze_device_cache_properties_t ze_device_cache_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_image_properties_t
-typedef struct _ze_device_image_properties_t ze_device_image_properties_t;
+//typedef struct _ze_device_image_properties_t ze_device_image_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_external_memory_properties_t
-typedef struct _ze_device_external_memory_properties_t ze_device_external_memory_properties_t;
+//typedef struct _ze_device_external_memory_properties_t ze_device_external_memory_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_p2p_properties_t
-typedef struct _ze_device_p2p_properties_t ze_device_p2p_properties_t;
+//typedef struct _ze_device_p2p_properties_t ze_device_p2p_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_context_desc_t
-typedef struct _ze_context_desc_t ze_context_desc_t;
+//typedef struct _ze_context_desc_t ze_context_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_command_queue_desc_t
-typedef struct _ze_command_queue_desc_t ze_command_queue_desc_t;
+//typedef struct _ze_command_queue_desc_t ze_command_queue_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_command_list_desc_t
-typedef struct _ze_command_list_desc_t ze_command_list_desc_t;
+//typedef struct _ze_command_list_desc_t ze_command_list_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_copy_region_t
-typedef struct _ze_copy_region_t ze_copy_region_t;
+//typedef struct _ze_copy_region_t ze_copy_region_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_image_region_t
-typedef struct _ze_image_region_t ze_image_region_t;
+//typedef struct _ze_image_region_t ze_image_region_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_event_pool_desc_t
-typedef struct _ze_event_pool_desc_t ze_event_pool_desc_t;
+//typedef struct _ze_event_pool_desc_t ze_event_pool_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_event_desc_t
-typedef struct _ze_event_desc_t ze_event_desc_t;
+//typedef struct _ze_event_desc_t ze_event_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_kernel_timestamp_data_t
-typedef struct _ze_kernel_timestamp_data_t ze_kernel_timestamp_data_t;
+//typedef struct _ze_kernel_timestamp_data_t ze_kernel_timestamp_data_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_kernel_timestamp_result_t
-typedef struct _ze_kernel_timestamp_result_t ze_kernel_timestamp_result_t;
+//typedef struct _ze_kernel_timestamp_result_t ze_kernel_timestamp_result_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_fence_desc_t
-typedef struct _ze_fence_desc_t ze_fence_desc_t;
+//typedef struct _ze_fence_desc_t ze_fence_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_image_format_t
-typedef struct _ze_image_format_t ze_image_format_t;
+//typedef struct _ze_image_format_t ze_image_format_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_image_desc_t
-typedef struct _ze_image_desc_t ze_image_desc_t;
+//typedef struct _ze_image_desc_t ze_image_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_image_properties_t
-typedef struct _ze_image_properties_t ze_image_properties_t;
+//typedef struct _ze_image_properties_t ze_image_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_device_mem_alloc_desc_t
-typedef struct _ze_device_mem_alloc_desc_t ze_device_mem_alloc_desc_t;
+//typedef struct _ze_device_mem_alloc_desc_t ze_device_mem_alloc_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_host_mem_alloc_desc_t
-typedef struct _ze_host_mem_alloc_desc_t ze_host_mem_alloc_desc_t;
+//typedef struct _ze_host_mem_alloc_desc_t ze_host_mem_alloc_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_memory_allocation_properties_t
-typedef struct _ze_memory_allocation_properties_t ze_memory_allocation_properties_t;
+//typedef struct _ze_memory_allocation_properties_t ze_memory_allocation_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_external_memory_export_desc_t
-typedef struct _ze_external_memory_export_desc_t ze_external_memory_export_desc_t;
+//typedef struct _ze_external_memory_export_desc_t ze_external_memory_export_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_external_memory_import_fd_t
-typedef struct _ze_external_memory_import_fd_t ze_external_memory_import_fd_t;
+//typedef struct _ze_external_memory_import_fd_t ze_external_memory_import_fd_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_external_memory_export_fd_t
-typedef struct _ze_external_memory_export_fd_t ze_external_memory_export_fd_t;
+//typedef struct _ze_external_memory_export_fd_t ze_external_memory_export_fd_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_module_constants_t
-typedef struct _ze_module_constants_t ze_module_constants_t;
+//typedef struct _ze_module_constants_t ze_module_constants_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_module_desc_t
-typedef struct _ze_module_desc_t ze_module_desc_t;
+//typedef struct _ze_module_desc_t ze_module_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_module_properties_t
-typedef struct _ze_module_properties_t ze_module_properties_t;
+//typedef struct _ze_module_properties_t ze_module_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_kernel_desc_t
-typedef struct _ze_kernel_desc_t ze_kernel_desc_t;
+//typedef struct _ze_kernel_desc_t ze_kernel_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_kernel_uuid_t
-typedef struct _ze_kernel_uuid_t ze_kernel_uuid_t;
+//typedef struct _ze_kernel_uuid_t ze_kernel_uuid_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_kernel_properties_t
-typedef struct _ze_kernel_properties_t ze_kernel_properties_t;
+//typedef struct _ze_kernel_properties_t ze_kernel_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_group_count_t
-typedef struct _ze_group_count_t ze_group_count_t;
+//typedef struct _ze_group_count_t ze_group_count_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_raytracing_mem_alloc_ext_desc_t
-typedef struct _ze_raytracing_mem_alloc_ext_desc_t ze_raytracing_mem_alloc_ext_desc_t;
+//typedef struct _ze_raytracing_mem_alloc_ext_desc_t ze_raytracing_mem_alloc_ext_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_sampler_desc_t
-typedef struct _ze_sampler_desc_t ze_sampler_desc_t;
+//typedef struct _ze_sampler_desc_t ze_sampler_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare ze_physical_mem_desc_t
-typedef struct _ze_physical_mem_desc_t ze_physical_mem_desc_t;
+//typedef struct _ze_physical_mem_desc_t ze_physical_mem_desc_t;
 
 
 #if !defined(__GNUC__)
 #pragma endregion
@@ -8914,5 +8914,5 @@
 #if defined(__cplusplus)
 } // extern "C"
 #endif
 
-#endif // _ZE_API_H
\ No newline at end of file
+#endif // _ZE_API_H
diff -u4 -r include/zet_api.h modified_include/zet_api.h
--- include/zet_api.h	2020-08-03 15:26:37.052772417 -0500
+++ modified_include/zet_api.h	2020-08-03 15:51:05.475513580 -0500
@@ -147,81 +147,81 @@
 } zet_typed_value_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_base_properties_t
-typedef struct _zet_base_properties_t zet_base_properties_t;
+//typedef struct _zet_base_properties_t zet_base_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_base_desc_t
-typedef struct _zet_base_desc_t zet_base_desc_t;
+//typedef struct _zet_base_desc_t zet_base_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_typed_value_t
-typedef struct _zet_typed_value_t zet_typed_value_t;
+//typedef struct _zet_typed_value_t zet_typed_value_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_device_debug_properties_t
-typedef struct _zet_device_debug_properties_t zet_device_debug_properties_t;
+//typedef struct _zet_device_debug_properties_t zet_device_debug_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_debug_config_t
-typedef struct _zet_debug_config_t zet_debug_config_t;
+//typedef struct _zet_debug_config_t zet_debug_config_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_debug_event_info_detached_t
-typedef struct _zet_debug_event_info_detached_t zet_debug_event_info_detached_t;
+//typedef struct _zet_debug_event_info_detached_t zet_debug_event_info_detached_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_debug_event_info_module_t
-typedef struct _zet_debug_event_info_module_t zet_debug_event_info_module_t;
+//typedef struct _zet_debug_event_info_module_t zet_debug_event_info_module_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_debug_event_info_thread_stopped_t
-typedef struct _zet_debug_event_info_thread_stopped_t zet_debug_event_info_thread_stopped_t;
+//typedef struct _zet_debug_event_info_thread_stopped_t zet_debug_event_info_thread_stopped_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_debug_event_t
-typedef struct _zet_debug_event_t zet_debug_event_t;
+//typedef struct _zet_debug_event_t zet_debug_event_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_debug_memory_space_desc_t
-typedef struct _zet_debug_memory_space_desc_t zet_debug_memory_space_desc_t;
+//typedef struct _zet_debug_memory_space_desc_t zet_debug_memory_space_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_debug_regset_properties_t
-typedef struct _zet_debug_regset_properties_t zet_debug_regset_properties_t;
+//typedef struct _zet_debug_regset_properties_t zet_debug_regset_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_metric_group_properties_t
-typedef struct _zet_metric_group_properties_t zet_metric_group_properties_t;
+//typedef struct _zet_metric_group_properties_t zet_metric_group_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_metric_properties_t
-typedef struct _zet_metric_properties_t zet_metric_properties_t;
+//typedef struct _zet_metric_properties_t zet_metric_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_metric_streamer_desc_t
-typedef struct _zet_metric_streamer_desc_t zet_metric_streamer_desc_t;
+//typedef struct _zet_metric_streamer_desc_t zet_metric_streamer_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_metric_query_pool_desc_t
-typedef struct _zet_metric_query_pool_desc_t zet_metric_query_pool_desc_t;
+//typedef struct _zet_metric_query_pool_desc_t zet_metric_query_pool_desc_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_profile_properties_t
-typedef struct _zet_profile_properties_t zet_profile_properties_t;
+//typedef struct _zet_profile_properties_t zet_profile_properties_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_profile_free_register_token_t
-typedef struct _zet_profile_free_register_token_t zet_profile_free_register_token_t;
+//typedef struct _zet_profile_free_register_token_t zet_profile_free_register_token_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_profile_register_sequence_t
-typedef struct _zet_profile_register_sequence_t zet_profile_register_sequence_t;
+//typedef struct _zet_profile_register_sequence_t zet_profile_register_sequence_t;
 
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Forward-declare zet_tracer_exp_desc_t
-typedef struct _zet_tracer_exp_desc_t zet_tracer_exp_desc_t;
+//typedef struct _zet_tracer_exp_desc_t zet_tracer_exp_desc_t;
 
 
 #if !defined(__GNUC__)
 #pragma endregion
@@ -1644,5 +1644,5 @@
 #if defined(__cplusplus)
 } // extern "C"
 #endif
 
-#endif // _ZET_API_H
\ No newline at end of file
+#endif // _ZET_API_H
