diff -u4 -r --new-file include/mpi.h modified_include/mpi.h
--- include/mpi.h      2024-06-18 15:52:35.000000000 +0000
+++ modified_include/mpi.h     2024-06-14 16:04:06.000000000 +0000
@@ -1,7 +1,7 @@
 #ifndef MPI_H_ABI
 #define MPI_H_ABI
 
-#include <stdint.h>
+//#include <stdint.h>
 
 #if defined(__cplusplus)
 extern "C" {
@@ -571,9 +571,9 @@ enum {
     MPI_T_PVAR_CLASS_GENERIC           =  10
 };
 
-typedef void (MPI_T_event_cb_function)(MPI_T_event_instance event_instance, MPI_T_event_registration event_registration, MPI_T_cb_safety cb_safety, void *user_data);
-typedef void (MPI_T_event_free_cb_function)(MPI_T_event_registration event_registration, MPI_T_cb_safety cb_safety, void *user_data);
-typedef void (MPI_T_event_dropped_cb_function)(MPI_Count count, MPI_T_event_registration event_registration, int source_index, MPI_T_cb_safety cb_safety, void *user_data);
+//typedef void (MPI_T_event_cb_function)(MPI_T_event_instance event_instance, MPI_T_event_registration event_registration, MPI_T_cb_safety cb_safety, void *user_data);
+//typedef void (MPI_T_event_free_cb_function)(MPI_T_event_registration event_registration, MPI_T_cb_safety cb_safety, void *user_data);
+//typedef void (MPI_T_event_dropped_cb_function)(MPI_Count count, MPI_T_event_registration event_registration, int source_index, MPI_T_cb_safety cb_safety, void *user_data);
 
 /* MPI global variables */
 extern MPI_Fint* MPI_F_STATUS_IGNORE;
@@ -1224,12 +1224,12 @@ int MPI_T_event_get_num(int *num_events);
 int MPI_T_event_get_source(MPI_T_event_instance event_instance, int *source_index);
 int MPI_T_event_get_timestamp(MPI_T_event_instance event_instance, MPI_Count *event_timestamp);
 int MPI_T_event_handle_alloc(int event_index, void *obj_handle, MPI_Info info, MPI_T_event_registration *event_registration);
-int MPI_T_event_handle_free(MPI_T_event_registration event_registration, void *user_data, MPI_T_event_free_cb_function free_cb_function);
+//int MPI_T_event_handle_free(MPI_T_event_registration event_registration, void *user_data, MPI_T_event_free_cb_function free_cb_function);
 int MPI_T_event_handle_get_info(MPI_T_event_registration event_registration, MPI_Info *info_used);
 int MPI_T_event_handle_set_info(MPI_T_event_registration event_registration, MPI_Info info);
 int MPI_T_event_read(MPI_T_event_instance event_instance, int element_index, void *buffer);
-int MPI_T_event_register_callback(MPI_T_event_registration event_registration, MPI_T_cb_safety cb_safety, MPI_Info info, void *user_data, MPI_T_event_cb_function event_cb_function);
-int MPI_T_event_set_dropped_handler(MPI_T_event_registration event_registration, MPI_T_event_dropped_cb_function dropped_cb_function);
+//int MPI_T_event_register_callback(MPI_T_event_registration event_registration, MPI_T_cb_safety cb_safety, MPI_Info info, void *user_data, MPI_T_event_cb_function event_cb_function);
+//int MPI_T_event_set_dropped_handler(MPI_T_event_registration event_registration, MPI_T_event_dropped_cb_function dropped_cb_function);
 int MPI_T_finalize(void);
 int MPI_T_init_thread(int required, int *provided);
 int MPI_T_pvar_get_index(const char *name, int var_class, int *pvar_index);
@@ -1892,12 +1892,12 @@ int PMPI_T_event_get_num(int *num_events);
 int PMPI_T_event_get_source(MPI_T_event_instance event_instance, int *source_index);
 int PMPI_T_event_get_timestamp(MPI_T_event_instance event_instance, MPI_Count *event_timestamp);
 int PMPI_T_event_handle_alloc(int event_index, void *obj_handle, MPI_Info info, MPI_T_event_registration *event_registration);
-int PMPI_T_event_handle_free(MPI_T_event_registration event_registration, void *user_data, MPI_T_event_free_cb_function free_cb_function);
+//int PMPI_T_event_handle_free(MPI_T_event_registration event_registration, void *user_data, MPI_T_event_free_cb_function free_cb_function);
 int PMPI_T_event_handle_get_info(MPI_T_event_registration event_registration, MPI_Info *info_used);
 int PMPI_T_event_handle_set_info(MPI_T_event_registration event_registration, MPI_Info info);
 int PMPI_T_event_read(MPI_T_event_instance event_instance, int element_index, void *buffer);
-int PMPI_T_event_register_callback(MPI_T_event_registration event_registration, MPI_T_cb_safety cb_safety, MPI_Info info, void *user_data, MPI_T_event_cb_function event_cb_function);
-int PMPI_T_event_set_dropped_handler(MPI_T_event_registration event_registration, MPI_T_event_dropped_cb_function dropped_cb_function);
+//int PMPI_T_event_register_callback(MPI_T_event_registration event_registration, MPI_T_cb_safety cb_safety, MPI_Info info, void *user_data, MPI_T_event_cb_function event_cb_function);
+//int PMPI_T_event_set_dropped_handler(MPI_T_event_registration event_registration, MPI_T_event_dropped_cb_function dropped_cb_function);
 int PMPI_T_finalize(void);
 int PMPI_T_init_thread(int required, int *provided);
 int PMPI_T_pvar_get_index(const char *name, int var_class, int *pvar_index);
